### PR TITLE
⚡Project Loading: Improved "Recovery File Found" Dialogue

### PR DESCRIPTION
**Pull request by ghost**
⚡️Improved "Project Recovery File Found" Dialogue
It now only shows if recovery file has different contents than main file,
Previously, it showed the dialogue if recovery file existed, so even if the recovery file was the same as the main, it would show an annoying dialogue.
But now it only shows the dialogue if there are changes in the recovery file!

**Changes proposed in this pull request:**

- Opening a project that has a recovery file with the same contents as the main file will no longer show a dialogue asking which file to load, and just automatically loads the main file
- Like before, opening a project with a recovery file containing new changes will show a dialogue, however unlike before if the recovery file was made at a newer time but doesn't have any changes, it will not show a dialogue

**Time to ping the Great, Almighty, Emperor of ct.js: @CosmoMyzrailGorynych**

### CODE REVIEW

**Review by ghost — COMMENTED** [review]
Also, @QuickBanjo asked me to ping them on this pr

**Comment by QuickBanjo** [comment]
tldr; **Recommend Reject**. Requesting review of this comment by @SaberTooth-Studios and @CosmoMyzrailGorynych.

## What problem is this solution addressing?

My understanding is this PR addresses an issue brought up by "antagonist" aka Dmitrii Z.#5109 on Discord:
>I have a dream - ct.js stopped creating recovery files that fully match original file and won't suggest me restore my work 'cause recovery file actually the same as original...
>[this occurs] both before before 1.7.0 and after updating to 1.7.0

This issue may be present on `v1.x`, but I am **Unable to Reproduce** this on `develop`. In particular, recovery files don't even appear to get created until changes are made. I verified this via the following two methods:

Basic:
1. Open a project.
2. Leave project open for at least 5 minutes (recovery save timeout) while taking no further action.
3. Verify that a .recovery file is not created alongside the project file.

Edit-Revert:
1. Open a project.
2. Make an easily reversible change, e.g. open an existing asset and click a checkbox, then Apply/Done.
3. Make another change "undoing" the change in step 2, e.g. unchecking the checkbox, then clicking Apply/Done.
4. Leave project open for at least 5 minutes (recovery save timeout) while taking no further action.
5. Verify that a .recovery file is not created alongside the project file.

If you _don't_ revert the change in the latter case, it appears to create the recovery file, as expected.

However, under the above conditions, there's no recovery file, and upon loading the project again, there's no recovery prompt.

### Edge Case

A possible edge case would be if we make a change, wait for recovery save, revert the change, then wait for recovery save again. Then we may have a recovery file that corresponds to the main project file. I have not attempted to reproduce this phenomenon on `develop`.

However, it's not clear if we even care in this scenario. I believe it's industry standard (e.g. Microsoft Word, etc.) _not_ to ignore or delete the recovery file in this case. Consequently, may want to keep this behavior.

But if we _do_ want to fix this...

## This PR

This appear to technically resolve the issue by loading the entire project and its recovery file from disk, and comparing before prompting to load the recovery file.

However, most of the performance loss in this area comes from loading files from disk, and this PR would effectively _triple_ load time, both worst case and expected average. Although this might not be noticeable on small projects on machines with fast disk read times, other scenarios e.g. a larger project on a hard disk drive will see huge (absolute) increases in load times. This solution reduces the scalability and platform-independence of loading performance.

Even if this PR fixes the aforementioned edge case, these performance issues probably don't outweigh this minor benefit. In particular, the benefit is intermittent reduction in loading times by 2-4 seconds (user isn't shown the recovery prompt), but the increased loading times could be arbitrarily large, perhaps even over a minute under certain conditions.

I **Recommend Reject** until these issues are resolved, or an alternative approach is taken.

## Backporting

Since the original issue may be present on `v1.x`, a solution analogous to this one may be worthwhile on that branch. Whether this is worthwhile depends on the ambition of the developer doing this, and the expected development schedule for v2.x.

**Comment by ghost** [comment]
@QuickBanjo I agree with your points, I have closed this pr and am finding a better, more optimized, approach to fixing it! Thanks for your feedback!
